### PR TITLE
[location][ios] Fix location background mode was required to use geofencing

### LINKED ISSUE #5054: Determine if we can skip background mode check when using Geofencing
Related post: https://forums.expo.io/t/ios-app-store-rejections-with-the-location-background-mode/25537/2

If possible, we should remove the check for background mode enabled if the user is just using the Geofencing API

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Fixes #5054 

# How

I removed checks for `location` background mode when starting/stoping geofencing.

# Test Plan

Should work 😅

### CODE REVIEW

File: ios/versioned-react-native/ABI33_0_0/EXLocation/ABI33_0_0EXLocation/ABI33_0_0EXLocation.m [hunk 1]
```diff
@@ -449,12 +461,17 @@ - (BOOL)checkPermissions:(ABI33_0_0UMPromiseRejectBlock)reject
   return YES;
 }
 
-- (BOOL)checkBackgroundServices:(ABI33_0_0UMPromiseRejectBlock)reject
+- (BOOL)checkTaskManagerExists:(ABI33_0_0UMPromiseRejectBlock)reject
```
Comment (sjchmiela) [inline ios/versioned-react-native/ABI33_0_0/EXLocation/ABI33_0_0EXLocation/ABI33_0_0EXLocation.m]:
```suggestion
- (BOOL)doesTaskManagerExist:(ABI33_0_0UMPromiseRejectBlock)reject
```

File: ios/versioned-react-native/ABI33_0_0/EXLocation/ABI33_0_0EXLocation/ABI33_0_0EXLocation.m [hunk 2]
```diff
@@ -449,12 +461,17 @@ - (BOOL)checkPermissions:(ABI33_0_0UMPromiseRejectBlock)reject
...
 {
   if (_tasksManager == nil) {
     reject(@"E_TASKMANAGER_NOT_FOUND", @"`expo-task-manager` module is required to use background services.", nil);
     return NO;
   }
+  return YES;
+}
+
+- (BOOL)checkBackgroundServices:(ABI33_0_0UMPromiseRejectBlock)reject
```
Comment (sjchmiela) [inline ios/versioned-react-native/ABI33_0_0/EXLocation/ABI33_0_0EXLocation/ABI33_0_0EXLocation.m]:
```suggestion
- (BOOL)areBackgroundServicesConfigured:(ABI33_0_0UMPromiseRejectBlock)reject
```

**Review by sjchmiela — COMMENTED** [review]
I would rather use this wording, but I'm ok with merging this as it is

**Review by sjchmiela — APPROVED** [review]

**Comment by zacksayenko** [comment]
@tsapeta Does this change mean that GeoFencing will not work in the background? What about when the app is killed?

**Pull request merged**